monai/utils/type_conversion.py

```diff
@@ -319,21 +275,11 @@ def convert_data_type(
 
     dtype_ = get_equivalent_dtype(dtype, output_type)
 
-    if not drop_meta and not issubclass(output_type, monai.data.MetaObj) and isinstance(data, monai.data.MetaObj):
-        # input has a MetaObj, user chose keep the metadata, but the output type cannot take a MetaObj.
-        if issubclass(output_type, torch.Tensor):
-            # user-specified MetaTensor to torch tensor keep the MetaTensor type, for backward compatibility
-            output_type = type(data)  # type: ignore
-        else:
-            raise RuntimeError(f"the specified output_type {output_type} cannot have the metaobj, but drop_meta=False.")
-
     data_: NdarrayTensor
 
-    if issubclass(output_type, monai.data.MetaTensor):
-        data_ = convert_to_meta_tensor(data, dtype=dtype_, device=device, wrap_sequence=wrap_sequence)
-        return data_, orig_type, orig_device
     if issubclass(output_type, torch.Tensor):
-        data_ = convert_to_tensor(data, dtype=dtype_, device=device, wrap_sequence=wrap_sequence)
+        track_meta = True if issubclass(output_type, monai.data.MetaTensor) else False
```

Comment:
this could be:
```suggestion
        track_meta = issubclass(output_type, monai.data.MetaTensor)
```

Reply:
Updated in the latest commit.

Thanks.